tavern_server/src/db_tables.sql

```diff
@@ -0,0 +1,861 @@
+        memory      		INT 
+		NOT NULL,
+        threads     		INT 
+		NOT NULL
+);
+
+--Character overview table
+CREATE TABLE IF NOT EXISTS Characters (
+        char_id         	UUID PRIMARY KEY,
+
+	id			UUID REFERENCES Users 
+		NOT NULL,
+        race_id         	UUID REFERENCES Races 
+		NOT NULL,
+        deity_id        	UUID REFERENCES Deities 
+		NOT NULL,
+
+        name            	TEXT 
+		NOT NULL,
+        age             	INT 
+		NOT NULL 
+		CHECK (age > 0),
+        gender          	Gender,
+        alignment       	Alignment 
+		NOT NULL,
+        backstory       	TEXT,
+        height          	INT,
+        weight          	INT,
+        size            	Size 
+		NOT NULL,
+
+        strength        	INT 
+		NOT NULL 
+		CHECK (strength >= 0 AND strength <= 20),
+        dexterity       	INT 
+		NOT NULL 
+		CHECK (dexterity >= 0 AND dexterity <= 20),
+        constitution    	INT 
+		NOT NULL
+		CHECK (constitution >= 0 AND constitution <= 20),
+        intelligence    	INT 
+		NOT NULL
+		CHECK (intelligence >= 0 AND intelligence <= 20),
+        wisdom          	INT
+		NOT NULL
+		CHECK (wisdom >= 0 AND wisdom <= 20),
+        charisma        	INT
+		NOT NULL
+		CHECK (charisma >= 0 AND charisma <= 20),
+
+        max_hp          	INT
+		NOT NULL
+		CHECK (max_hp >= 0),
+        damage          	INT
+		NOT NULL
+		CHECK (damage >= 0),
+        nonlethal       	INT
+		NOT NULL
+		CHECK (nonlethal >= 0),
+
+        copper          	INT
+		NOT NULL
+		CHECK (copper >= 0),
+        silver          	INT
+		NOT NULL
+		CHECK (silver >= 0),
+        gold            	INT
+		NOT NULL
+		CHECK (gold >= 0),
+        platinum        	INT
+		NOT NULL
+		CHECK (platinum >= 0)
+);
+
+--Race tables
+CREATE TABLE IF NOT EXISTS Races (
+	race_id 		UUID PRIMARY KEY,
+	type_id			UUID REFERENCES RaceTypes
+		NOT NULL,
+	subtype_id		UUID REFERENCES RaceSubtypes,
+
+	name			TEXT
+		NOT NULL,
+	move_speed		INT
+		NOT NULL,
+	size			Size
+		NOT NULL,
+	languages		TEXT
+);
+
+CREATE TABLE IF NOT EXISTS RaceTypes (
+	type_id			UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL,
+	hit_die			TEXT
+		NOT NULL,
+	bab_per_hit_die		REAL
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXiSTS RaceSubtypes (
+	subtype_id		UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL
+);
+	
+--Class tables
+CREATE TABLE IF NOT EXISTS Classes (
+	class_id		UUID PRIMARY KEY,
+
+	name			TEXT
+		NOT NULL,
+	hit_die			TEXT
+		NOT NULL,
+	starting_wealth		TEXT
+		NOT NULL,
+	bab_per_level		REAL
+		NOT NULL,
+	skills_per_level	INT
+		NOT NULL,
+	skills_attr		Attribute
+		NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS Subclasses (
+	subclass_id		UUID PRIMARY KEY,
+	class_id		UUID REFERENCES Classes
+		NOT NULL,
+
+	caster_type		CasterType,
+	casting_attr		Attribute
+);
+
+CREATE TABLE IF NOT EXISTS characterSubclasses (
+	char_id			UUID REFERENCES Characters
+		NOT NULL,
+	subclass_id		UUID REFERENCES Subclasses
+		NOT NULL,
+
+	levels_taken		INT
+		NOT NULL
+		CHECK (levels_taken > 0),
```

Comment:
We should discuss if it is possible to have a level 0 character, e.g. one saved in the middle of creation.

Reply:
I agree. I know a lot of people do it piecemeal, but it does mean letting a lot of columns be nullable, or link to default "empty" classes and races and spells. I don't think that's ideal, admittedly, but I'm open to being persuaded.